bigframes/ml/decomposition.py

```diff
@@ -197,3 +206,163 @@ def score(
 
         # TODO(b/291973741): X param is ignored. Update BQML supports input in ML.EVALUATE.
         return self._bqml_model.evaluate()
+
+
+@log_adapter.class_logger
+class MatrixFactorization(
+    base.UnsupervisedTrainablePredictor,
+    bigframes_vendored.sklearn.decomposition._mf.MatrixFactorization,
+):
+    __doc__ = bigframes_vendored.sklearn.decomposition._mf.MatrixFactorization.__doc__
+
+    def __init__(
+        self,
+        *,
+        feedback_type: Literal["explicit", "implicit"] = "explicit",
+        num_factors: int,
+        user_col: str,
+        item_col: str,
+        rating_col: str = "rating",
+        # TODO: Add support for hyperparameter tuning.
+        l2_reg: float = 1.0,
+    ):
+
+        if feedback_type not in ("explicit", "implicit", "EXPLICIT", "IMPLICIT"):
+            raise ValueError(
+                f"Expected feedback_type to be `explicit` or `implicit`, but got {feedback_type}"
+            )
```

Comment:
I would suggest making this case-insensitive, instead.

```suggestion
        feedback_type = feedback_type.lower()
        if feedback_type not in ("explicit", "implicit"):
            raise ValueError(
                f"Expected feedback_type to be `explicit` or `implicit`, but got {feedback_type}"
            )
```